doc/howtos/rdtraining/sprinkles.rst

```diff
@@ -0,0 +1,485 @@
+`action <https://github.com/odoo/odoo/blob/6decc32a889b46947db6dd4d42ef995935894a2a/addons/crm/report/crm_opportunity_report_views.xml#L115>`__
+with the
+`corresponding filter <https://github.com/odoo/odoo/blob/6decc32a889b46947db6dd4d42ef995935894a2a/addons/crm/report/crm_opportunity_report_views.xml#L68>`__.
+
+.. exercise:: Add a default filter
+
+    Make the 'Available' filter selected by default on the ``estate.property`` action.
+
+Another useful improvement in our module would be to be able to search efficiently by living area.
+In practice, a user will want to search for properties of 'at least' the given area. There is no
+real use case where one would want to find a property of an exact living area. It is always
+possible to make a custom search, but that's not convenient. 
+
+Search view ``<field>`` elements can have a ``filter_domain`` that overrides
+the domain generated for searching on the given field. In the given domain,
+``self`` represents the value entered by the user. In the example below, it is
+used to search on both fields ``name`` and ``description``.
+
+.. code-block:: xml
+
+    <search string="Test">
+        <field name="description" string="Name and description"
+               filter_domain="['|', ('name', 'ilike', self), ('description', 'ilike', self)]"/>
+        </group>
+    </search>
+
+.. exercise:: Change living area search
+
+    Add a ``filter_domain`` on the living area to include properties with an area larger than the
+    given value.
+
+Stat Buttons
+============
+
+.. note::
+
+    **Goal**: at the end of this section, a stat button on the property type which gives the list
+    of all offers related to properties of the given type.
+
+    .. image:: sprinkles/media/stat_button.gif
+      :align: center
+      :alt: Stat button
+
+If you already used some functional modules in Odoo, you probably already encountered a 'stat
+button'. These buttons are displayed on the top right of a form view and give a quick access to
+linked documents. In our real estate module, we would like to have a quick link to the offers
+related to a given property type, as depicted in the **Goal** of the section.
+
+At this point of the tutorial we have seen mostly all the concepts to be able to do it. However,
+there is not a single solution, and it can still be confusing if you don't know where to start from.
+We'll describe a step-by-step solution in the exercise. It can always be useful to find some
+examples in the Odoo codebase by looking for ``oe_stat_button``.
+
+The following exercise might be a bit more difficult than the previous ones since it assumes you
+are able to search examples in the source code on your own. If you are stuck, there is probably
+someone close to you who can help you ;-)
+
+.. exercise:: Add a stat button on property type
+
+    - Add the field ``property_type_id`` on ``estate.property.offer``. We can define it as a
+      related field on ``property_id.property_type_id`` and set as stored.
```

Comment:
Correct me if I'm wrong, but I think it's the first time that we talk about  'related field', maybe we can make a note and explain what is a related field, and added an exercise in the relations chapter.

Reply:
Explanation added. A related field is more a computed field actually.